src/Stratis.Bitcoin.Features.Consensus/Rules/ProvenHeaderRules/ProvenHeaderSizeRule.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Consensus.Rules;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.ProvenHeaderRules
+{
+    /// <summary>
+    /// Rule to check if the serialized sizes of the proven header components, such as merkle proof (max 512 bytes),
+    /// signature (max 80 bytes) and coinstake (max 1,000,000 bytes), do not exceed maximum possible size allocation.
+    /// </summary>
+    /// <seealso cref="ProvenHeaderValidationConsensusRuleBase" />
+    public class ProvenHeaderSizeRule : ProvenHeaderValidationConsensusRuleBase
+    {
+        public override void Run(RuleContext context)
+        {
+            Guard.NotNull(context.ValidationContext.ChainedHeaderToValidate, nameof(context.ValidationContext.ChainedHeaderToValidate));
+
+            int height = context.ValidationContext.ChainedHeaderToValidate.Height;
+
+            if (context.SkipValidation || !this.IsProvenHeaderActivated(height))
```

Comment:
I suggest you split this in two classes `ProvenHeaderSizeRule` and `StratisProvenHeaderSizeRule`.
The later will have the check `IsProvenHeaderActivated` and then call the base if yes.

Reply:
Problem is that you can't have double inheritance. i.e. if `StratisProvenHeaderSizeRule` inherits from `ProvenHeaderSizeRule` it cannot then also inherit a `StratisProvenHeaderRuleBase` (the one you want to keep separate with a `IsProvenHeaderActivated` method).

Reply:
I can just rename 
`ProvenHeaderRuleBase` => `StratisProvenHeaderRuleBase`
and
`ProvenHeaderSizeRule` => `StratisProvenHeaderSizeRule`

If we ever have other POS implementations, we can action it then